Issue #56 - Allow filtering by changeset's description

armenzg opened the pull request:
This is useful if you want to see changesets associated to the
same bug.

f? @calixteman @marco-c

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/mozilla/firefox-code-coverage-frontend/184)
<!-- Reviewable:end -->

armenzg commented:
The deployment should show up in here soon:
https://firefox-code-coverage-pr-184.herokuapp.com/

Here's a screenshot until then:
![image](https://user-images.githubusercontent.com/44410/40385859-f8474f44-5dd5-11e8-91e1-5a2bf393e48b.png)

marco-c commented:
Looks good!

jankeromnes commented:
Thanks for implementing this!

A few thoughts:

- I'm not sure how many visitors would use this feature (e.g. you can already use your browser's Cmd+F to find particular commit descriptions)
- Given the above, maybe the filter UI shouldn't take up too much real estate on the front page. For example, you could hide it by default, and show a tiny [filter icon](https://materialdesignicons.com/icon/filter) (e.g. top right above the list of changesets, or right next to the `Description` column title) that reveals the form when you click it
- Also a few nits: Please add a non-breaking space between `:` and the `<input>`
- Additionally, maybe some fuzzy matching (like `fzf`) could be helpful too

armenzg commented:
The RelMan request was specifically being able to show changesets associated to a bug.
Since filtering by description was a more general implementation and a bigger win I decided to go for it.

I won't go as far as looking into `fzf` but I will implement the others.

armenzg commented:
Removing review request until I address the feedback.

armenzg commented:
Addressed feedback.

@jankeromnes r?

![image](https://user-images.githubusercontent.com/44410/40431996-9c056526-5e76-11e8-8821-94b1f133bba4.png)

armenzg commented:
r? @jankeromnes 
I've included your suggestions.

jankeromnes commented on src/containers/summary.jsx:
```diff
@@ -109,7 +117,8 @@ export default class SummaryContainer extends Component {
     const viewableCsetsMap = {};
     changesets.forEach((cset) => {
-      if (coverageMap[cset.node].show) {
+      if ((coverageMap[cset.node].show) &&
```
Nit: the double parens aren't really needed here.

jankeromnes commented on src/style.css:
```diff
@@ -158,6 +158,13 @@ tr.changeset td{
+}
+form:focus-within + button {
+  display: none;
+}
```
Nit: inconsistent indentation.
